recipes/libq/all/conanfile.py

```diff
@@ -0,0 +1,88 @@
+import os
+
+from conan import ConanFile
+from conan.errors import ConanInvalidConfiguration
+from conan.tools.build import check_min_cppstd
+from conan.tools.cmake import CMake, CMakeToolchain, cmake_layout
+from conan.tools.files import copy, get, rm, rmdir, export_conandata_patches, apply_conandata_patches
+from conan.tools.microsoft import is_msvc
+from conan.tools.scm import Version
+from conan.tools.apple import fix_apple_shared_install_name
+
+required_conan_version = ">=2.1"
+
+class libqConan(ConanFile):
+    name = "libq"
+    description = "A platform-independent promise library for C++, implementing asynchronous continuations."
+    license = "Apache-2.0"
+    topics = ("async", "promises")
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://github.com/grantila/q"
+    settings = "os", "compiler", "build_type", "arch"
+    package_type = "library"
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+    }
+    default_options = {
+        "shared": False,
+        "fPIC": True,
+    }
+
+    implements = ["auto_shared_fpic"]
+
+    def layout(self):
+        cmake_layout(self, src_folder="src")
+
+    def export_sources(self):
+        export_conandata_patches(self)
+
+    def validate(self):
+        check_min_cppstd(self, 11)
+        if self.settings.os == "Windows" and self.options.shared:
+            # It exports no symbols, so it can't be used as a shared library
+            raise ConanInvalidConfiguration("Does not support shared libraries on Windows")
```

Comment:
```suggestion
	def config_options(self):
	    if self.settings.os == "Windows":
	        del self.options.fPIC
	        self.package_type = "static-library"

    def validate(self):
        check_min_cppstd(self, 11)        
```

I would suggest enforcing Windows libs as static only.